### PR TITLE
initial selinux test

### PR DESCRIPTION
**Pull request by karabijavad**
excited to help out on this project

looking forward to any feedback

thank you

### CODE REVIEW

**Comment by karabijavad** [comment]
@metan-ucw 
well, i guess its only testing that it can setenforce(), so not too much is happening. i will add some more tests

thanks for all the input. i will make the changes you suggested

File: testcases/kernel/security/selinux/Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+top_srcdir		?= ../../../..
+
+include $(top_srcdir)/include/mk/testcases.mk
+
+INSTALL_TARGETS		:= testall.sh
+
```
Comment (metan-ucw) [inline testcases/kernel/security/selinux/Makefile]: This still references ```testall.sh``` that does not exists.

File: testcases/kernel/security/selinux/selinux.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+static int old_enforcing;
+
+static void setup(void)
+{
+	tst_require_root();
+	tst_tmpdir();
+
+	if (!is_selinux_enabled())
+		tst_brkm(TBROK, cleanup, "selinux not enabled\n");
+
```
Comment (metan-ucw) [inline testcases/kernel/security/selinux/selinux.c]: This is ```TCONF``` and not ```TBROK```. The ```TBROK``` means that some of the initialization steps has failed while ```TCONF``` means that test is not applicable.

File: testcases/kernel/security/selinux/selinux.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	old_enforcing = security_getenforce();
+
+	int setenforce_failure = security_setenforce(1);
+
+	if (setenforce_failure)
+		tst_brkm(TBROK, cleanup,
+			"unable to enforce selinux: %d", errno);
+
```
Comment (metan-ucw) [inline testcases/kernel/security/selinux/selinux.c]: If the test is testing that security_setenforce() could be called it shouldn't be done in the setup() and the test should do ```tst_resm(TPASS, ...)``` and ```tst_resm(TFAIL, ...)``` accordingly.

Also setenforce_failure is a poor name for a variable it's overly long and confusing, it should be simply named ```ret``` or ```res```.

File: testcases/kernel/security/selinux/selinux.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	setup();
+
+	int fd;
+
+	fd = creat("file", 0600);
+	if (-1 == fd)
+		tst_brkm(TBROK, cleanup, "could not create file: %d", errno);
+	else
+		printf("created file %d\n", fd);
+
```
Comment (metan-ucw) [inline testcases/kernel/security/selinux/selinux.c]: You must not use ```printf()``` in the testcases. We have ```tst_resm(TINFO, ...)``` for printing additional information.

Also I still do not understand what is the purpose of the creat() call.

If you want to write a test you need to think of test assertions first. In this case assertions should be derived from libselinux documentation. Looking at it the ```security_setenforce()``` just opens and writes the $selinux_mnt/enforce file so the easiest test can be:

* get enforce value from security_getenforce()
* inverse it and call security_setenforce()
* check that security_getenforce() returns new value
* set the old enforce value

More interesting testcase may do:

* Check that the setenforce was logged by auditd
* Check that policies are actually enforced when setenforce was set

File: testcases/kernel/security/selinux/selinux.c [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	tst_rmdir();
+	security_setenforce(old_enforcing);
+}
+
+int main(void)
+{
+	setup();
+
+	int fd;
+
```
Comment (metan-ucw) [inline testcases/kernel/security/selinux/selinux.c]: The test should pass argc and argv to the test option parser and should also do the standard looping.

**Comment by metan-ucw** [comment]
You should also merge the individual patches into one.

**Comment by karabijavad** [comment]
@metan-ucw 
i can not put into words how much i appreciate you looking over my pull request
i do not mean to waste your time on this, so i just want to let you know how grateful i am

i hope to soon be providing alot of good tests to LTP, and this is my first attempt, so i will learn from your suggestions and provide alot more tests with that knowledge

i will go over your comments and fix them

thank you

**Comment by metan-ucw** [comment]
Feel free to open new pull request if you proceed on the test.

**Pull request closed without merging**